lib/scholar/linear/ridge.ex

```diff
@@ -0,0 +1,324 @@
+
+  #{NimbleOptions.docs(@opts_schema)}
+
+  ## Return Values
+
+    The function returns a struct with the following parameters:
+
+    * `:coefficients` - Estimated coefficients for the linear regression problem.
+
+    * `:intercept` - Independent term in the linear model.
+
+  ## Examples
+
+      iex> x = Nx.tensor([[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]])
+      iex> y = Nx.tensor([4.0, 3.0, -1.0])
+      iex> Scholar.Linear.Ridge.fit(x, y)
+      %Scholar.Linear.Ridge{
+        coefficients: #Nx.Tensor<
+          f32[2]
+          [-0.42378732562065125, -0.6891375780105591]
+        >,
+        intercept: #Nx.Tensor<
+          f32
+          5.656937599182129
+        >
+      }
+  """
+  deftransform fit(a, b, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+
+    sample_weights? = opts[:sample_weights] != nil
+    kernel_cholesky = opts[:solver] == :cholesky and Nx.axis_size(a, 0) < Nx.axis_size(a, 1)
+
+    opts =
+      [
+        sample_weights_flag: sample_weights?,
+        rescale_flag: sample_weights? and not kernel_cholesky
+      ] ++
+        opts
+
+    {sample_weights, opts} = Keyword.pop(opts, :sample_weights, 1.0)
+    sample_weights = Nx.tensor(sample_weights)
+
+    {alpha, opts} = Keyword.pop!(opts, :alpha)
+    alpha = Nx.tensor(alpha) |> Nx.flatten()
+    num_targets = if Nx.rank(b) == 1, do: 1, else: Nx.axis_size(b, 1)
+
+    if Nx.size(alpha) not in [0, 1, num_targets] do
+      raise ArgumentError,
+            "expected number of targets be the same as number of penalties, got: #{inspect(num_targets)} != #{inspect(Nx.size(alpha))}"
+    end
+
+    fit_n(a, b, sample_weights, alpha, opts)
+  end
+
+  defnp fit_n(a, b, sample_weights, alpha, opts) do
+    flatten? = Nx.rank(b) == 1
+    num_targets = if Nx.rank(b) == 1, do: 1, else: Nx.axis_size(b, 1)
+
+    alpha =
+      if Nx.size(alpha) == 1 and num_targets > 1,
+        do: Nx.broadcast(alpha, {num_targets}),
+        else: alpha
+
+    {a_offset, b_offset} =
+      if opts[:fit_intercept?] do
+        preprocess_data(a, b, sample_weights, opts)
+      else
+        {_, a_shape} = Nx.shape(a)
+        b_reshaped = if Nx.rank(b) > 1, do: b, else: Nx.reshape(b, {:auto, 1})
+        {_, b_shape} = Nx.shape(b_reshaped)
+        {Nx.broadcast(0.0, {a_shape}), Nx.broadcast(0.0, {b_shape})}
+      end
+
+    {a, b} = {a - a_offset, b - b_offset}
+    {num_samples, num_features} = Nx.shape(a)
+
+    {a, b} =
+      if opts[:rescale_flag] do
+        rescale(a, b, sample_weights)
+      else
+        {a, b}
+      end
+
+    b = if Nx.rank(b) == 1, do: Nx.reshape(b, {:auto, 1}), else: b
+
+    coeff =
+      case opts[:solver] do
+        :cholesky ->
+          if num_samples >= num_features do
+            solve_cholesky(a, b, alpha)
+          else
+            kernel = Nx.dot(a, [1], a, [1])
+            dual_coeff = solve_cholesky_kernel(kernel, b, alpha, sample_weights, opts)
+            Nx.dot(a, [0], dual_coeff, [0]) |> Nx.transpose()
```

Comment:
Can't you swap the arguments and change contraction axes to avoid the transpose?